avocado/core/test.py

```diff
@@ -243,6 +244,10 @@ def record_and_warn(*args, **kwargs):
         self._stdout_file = os.path.join(self.logdir, 'stdout')
         self._stderr_file = os.path.join(self.logdir, 'stderr')
         self._output_file = os.path.join(self.logdir, 'output')
+        self._stdout_diff_file = os.path.join(self.logdir,
+                                              'stdout_expected.diff')
+        self._stderr_diff_file = os.path.join(self.logdir,
+                                              'stderr_expected.diff')
```

Comment:
These variables are only used in `_check_reference_stdout`, resp `_stderr` methods so I don't think it's necessary to store them globally (we already have way too many variables stored in `Test`)